daemon/container_unix.go

```diff
@@ -948,7 +948,8 @@ func (container *Container) connectToNetwork(idOrName string, updateSettings boo
 
 	ep, err := container.getEndpointInNetwork(n)
 	if err == nil {
-		return fmt.Errorf("container already connected to network %s", idOrName)
```

Comment:
Its not about uniqueness within the same host. It also impacts the uniqueness across hosts (though it is best effort). So, either we must completely remove it (which means we dont even perform the best effort) and create multiple endpoints of the same name. Or error out stating the fact.